File: src/0700-Search-in-a-Binary-Search-Tree/0700.cpp
class Solution {
 public:
  TreeNode *searchBST(TreeNode *root, int val) {
    if (root == nullptr) {
      return nullptr;
    }
    if (val == root->val) {
      return root;
    }
    return searchBST(val < root->val ? root->left : root->right, val);
  }
};

File: src/0700-Search-in-a-Binary-Search-Tree/0700.rs
use std::cell::RefCell;
use std::cmp::Ordering;
use std::rc::Rc;
impl Solution {
    pub fn search_bst(
        root: Option<Rc<RefCell<TreeNode>>>,
        val: i32,
    ) -> Option<Rc<RefCell<TreeNode>>> {
        if let Some(t) = root {
            let v = t.borrow().val;
            match v.cmp(&val) {
                Ordering::Equal => Some(t),
                Ordering::Greater => Self::search_bst(t.borrow_mut().left.take(), val),
                Ordering::Less => Self::search_bst(t.borrow_mut().right.take(), val),
            }
        } else {
            None
        }
    }
}
